src/test/java/core/CredentialsTest.java

```diff
@@ -1,55 +1,227 @@
+    
+    private static final String USERNAME_INPUT = "_.username";
+    private static final String PRIVATE_KEY_INPUT = "_.privateKey";
+    
+    private static final String CRED_ID = "ID";
+    private static final String CRED_USER = "user";
+    private static final String CRED_PWD = "password";
+    
     @Test @WithPlugins("ssh-credentials")
     public void createSshKeys() {
         final ManagedCredentials c = new ManagedCredentials(jenkins);
 
-        final String username = "xyz";
-        final String privKey = "1212121122121212";
-
         c.open();
         final SshPrivateKeyCredential sc = c.add(SshPrivateKeyCredential.class);
-        sc.username.set(username);
-        sc.selectEnterDirectly().privateKey.set(privKey);
+        sc.username.set(CRED_USER);
+        sc.selectEnterDirectly().privateKey.set(CRED_PWD);
+        sc.setId(CRED_ID);
         c.save();
 
         //now verify
-        jenkins.visit("credentials");
-        assertThat(find(by.input("_.username")).getAttribute("value"), is(equalTo(username)));
-        assertThat(find(by.input("_.privateKey")).getText(), is(equalTo(privKey)));
+        verifyInputValue(CRED_ID, USERNAME_INPUT, CRED_USER);
+        verifyInputValue(CRED_ID, PRIVATE_KEY_INPUT, CRED_PWD);
     }
 
     @Test
     public void createUserPwd() {
         final ManagedCredentials c = new ManagedCredentials(jenkins);
 
-        final String username = "xyz";
-        final String password = "1212121122121212";
-
         c.open();
         final UserPwdCredential upc = c.add(UserPwdCredential.class);
-        upc.username.set(username);
-        upc.password.set(password);
+        upc.username.set(CRED_USER);
+        upc.password.set(CRED_PWD);
+        upc.setId(CRED_ID);
         c.save();
 
         //now verify
         jenkins.visit("credentials");
-        assertThat(find(by.input("_.username")).getAttribute("value"), is(equalTo(username)));
+        verifyInputValue(CRED_ID, USERNAME_INPUT, CRED_USER);
+    }
+    
+    @Test
+    public void manageSystemScopedCredentialsTest() {
+        createUpdateDeleteTest(SYSTEM_SCOPE);
+    }
+
+    @Test
+    public void manageGlobalScopedCredentialsTest() {
+        createUpdateDeleteTest(GLOBAL_SCOPE);
+    }
+    
+    private void createUpdateDeleteTest(String systemScope) {
+        final ManagedCredentials c = new ManagedCredentials(jenkins);
+
+        // Create credential
+        c.open();
+        UserPwdCredential upc = c.add(UserPwdCredential.class);
+        upc.username.set(CRED_USER);
+        upc.password.set(CRED_PWD);
+        upc.description.set("Credential description");
+        upc.scope.select(systemScope);
+        upc.setId(CRED_ID);
+        c.save();
+
+        // verify credential was created
+        verifyInputValue(CRED_ID, USERNAME_INPUT, CRED_USER);
+        
+        // Update credential
+        c.open();
+        upc = c.get(UserPwdCredential.class, CRED_ID);
```

Comment:
Is it really needed to look up the `PageArea` like that (the implementation of the `get`methods sound fragile to me)? Why not reuse the instance you already have? The fact you navigated away and back should not matter.